Source/ASDisplayNode+Layout.mm

```diff
@@ -817,11 +817,53 @@ - (void)_completeLayoutTransition:(ASLayoutTransition *)layoutTransition
   }
 }
 
+- (void)_assertSubnodeState
+{
+  if (self.automaticallyManagesSubnodes == NO) {
+    return;
+  }
+
+  // Verify that any orphaned nodes are removed.
+  // This can occur in rare cases if main thread layout is flushed while a background layout is calculating.
+  NSArray<ASDisplayNode *> *subnodes = [self subnodes];
+  NSHashTable<ASDisplayNode *> *currentSubnodes =
+                                    [[NSHashTable alloc] initWithOptions:NSHashTableObjectPointerPersonality
+                                                                capacity:subnodes.count];
```

Comment:
Totally optional, you can do `auto currentSubnodes = [[NSHashTable<ASDisplayNode *> alloc]…]` to save some characters.